File: code/community/1306408472320937988/streaming-audio-language-detection.js
require('dotenv').config();
const WebSocket = require('ws');

const API_KEY = process.env.DEEPGRAM_API_KEY;
const ws = new WebSocket('wss://api.deepgram.com/v1/listen?model=multi', {
    headers: {
        'Authorization': `Token ${API_KEY}`
    }
});

ws.on('open', function open() {
    console.log('Connected to Deepgram server');
    
    // Replace with your own audio stream
    ws.send('Your audio stream chunk here');
});

ws.on('message', function incoming(data) {
    console.log(`Received: ${data}`);
});

// Ensure you have .env file with DEEPGRAM_API_KEY
// And replace 'Your audio stream chunk here' with actual data

File: code/community/1306408472320937988/streaming-audio-language-detection.go
package main

import (
    "bufio"
    "fmt"
    "log"
    "net/url"
    "os"
    "os/signal"
    "syscall"

    "github.com/gorilla/websocket"
    "github.com/joho/godotenv"
)

func main() {
    err := godotenv.Load()
    if err != nil {
        log.Fatal("Error loading .env file")
    }

    apiKey := os.Getenv("DEEPGRAM_API_KEY")
    u := url.URL{Scheme: "wss", Host: "api.deepgram.com", Path: "/v1/listen", RawQuery: "model=multi"}
    headers := make(map[string][]string)
    headers["Authorization"] = []string{"Token " + apiKey}

    c, _, err := websocket.DefaultDialer.Dial(u.String(), headers)
    if err != nil {
        log.Fatal("dial:", err)
    }
    defer c.Close()

    done := make(chan struct{})

    go func() {
        defer close(done)
        for {
            _, message, err := c.ReadMessage()
            if err != nil {
                log.Println("read:", err)
                return
            }
            fmt.Printf("Received: %s\n", message)
        }
    }()

    msg := "Your audio stream chunk here" // replace with actual audio data
    c.WriteMessage(websocket.TextMessage, []byte(msg))

    interrupt := make(chan os.Signal, 1)
    signal.Notify(interrupt, syscall.SIGINT, syscall.SIGTERM)

    <-interrupt

    fmt.Println("interrupt")

    // Cleanly close the connection by sending a close message and then
    // waiting (with timeout) for the server to close the connection.
    err = c.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
    if err != nil {
        log.Println("write close:", err)
        return
    }

    select {
    case <-done:
    case <-interrupt:
    }

    c.Close()
}

// Ensure .env file is present with DEEPGRAM_API_KEY
// Replace "Your audio stream chunk here" with actual data